docs/developer.rst

```diff
@@ -192,6 +192,35 @@ Example for an application, which should reside under ``src/modules/applications
             def save(self, path, iter_count):
                 save_your_application(self.application, f"{path}/application.txt")
 
+Writing an asynchronous Module
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+A typical example for an asynchronous module is a solver which submits its job into
+the queue of some server and retrieves the result some times later. In QUARK this is
+supported via the interrupt/resume mechanism.
+
+QUARK modules may return instructions to the benchmark manager as first entry in the return value of
+pre and post-process. Currently the following instructions are supported:
+    - PROCEED
+    - INTERRUPT
+
+PROCEED: if the benchmark manger gets a PROCEED (or no instruction at all) he continues with the regular QUARK work flow.
+If the job manager can finish the current job without getting an INTERRUPT or exception he adds
+"quark_job_status"=FINISHED to the metrics.
+
+INTERRUPT: if the benchmark manager gets an INTERRUPT he stops the current QUARK work flow,
```

Comment:
INTERRUPT: If the BenchmarkManager gets the instruction "INTERRUPT" it stops the current QUARK workflow,
adds "quark_job_status"=INTERRUPTED to the metrics, saves all the metrics written so far to the BenchmarkRecord
and continues with the configuration/repetition loop.